snap/validate.go

```diff
@@ -666,6 +665,15 @@ func ValidateApp(app *AppInfo) error {
 		}
 	}
 
+	// ActivatesOn is a list of slot names that use the "dbus" type
+	for _, slotName := range app.ActivatesOn {
+		if slot, ok := app.Slots[slotName]; !ok {
+			return fmt.Errorf("invalid activates-on value %q: slot not found", slotName)
+		} else if slot.Interface != "dbus" {
+			return fmt.Errorf("invalid activates-on value %q: slot does not use dbus interface", slotName)
```

Comment:
perhaps:

"..: slot interface is not of the currently supported type (dbus) for activation"

there's a chance will add more types that are supported in activates-on